### DISCUSSION TITLE
Improve error messaging on the 300 file git diff limit for triggering actions

### DISCUSSION THREAD

**Discussion by nipunn1313** [community/community#53831] [Actions] [answered]
### Select Topic Area

Bug

### Body

From
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore
```
Diffs are limited to 300 files. If there are files changed that aren't matched in the first 300 files returned by the filter, the workflow will not run. You may need to create more specific filters so that the workflow will run automatically.
```

Currently, github actions silently does not trigger the action run if it edits > 300 files and the file that matches the filter is > 300th one. The experience is very unexpected.

I have a few asks
1) Improve the error message in this scenario. Produce some kind of information showing that it was unable to properly calculate the diff and thus didn't kick off the run. It was very unclear what was going on.

2) Increase the limit (or make it configurable) - 300 simply seems low. If it's expensive on your end, maybe it could be higher on paid accounts.

Github is clearly capable of figuring out the number of files that changed - so having some way to get it to do the extra work would be nice.
![image](https://user-images.githubusercontent.com/1300387/234736840-a9f68ee7-0569-4689-a0a5-2f89e78163d4.png)
and listing them in the change list
![image](https://user-images.githubusercontent.com/1300387/234736923-c47c9a0c-1d31-4bd0-92a5-e8c1d8f03740.png)

I am open to other solutions! I realize I may be conveying the pain point in the form of a poorly constructed solution, since I don't understand the system well as well as the team at github.

I appreciate all the good work on github actions from github. I understand that error messages in corner cases are often not the priority, but in this case I suspect a small amount of error messaging could go a long way.

**Comment by jsoref** [comment]
The filter is clearly GitHub protecting itself from doing expensive work.

You can remove the filter from `on:` and implement it in a step. There's at least one action available to help with this case.

**Reply by jsoref** [reply]
Since people are still asking, I guess I should offer one possible implementation: https://github.com/marketplace/actions/paths-changes-filter

Note: this is *not* an endorsement.

**Comment by Armadillidiid** [comment]
Hey, I currently just faced the same issue in my GitHub action for a monorepo. How did you circumvent this?

**Comment by thyeggman** [comment, marked as answer]
@nipunn1313 I've reorganized the [docs](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#git-diff-comparisons) describing the limits to filter diffs after we recently made some improvements. The limit for number of files per diff is now 3000, up from 300. Hopefully the behavior is clearer now.

Unfortunately we can't do a lot about the error messages on workflows themselves, because as @jsoref mentioned, we are protecting our platform from expensive diff calculations. So when we get the 3000-file diff back, we can't know which files would have run if the diff were unlimited because that would require us to calculate the whole diff anyways.

**Reply by nipunn1313** [reply]
THX
